Application/SolutionScripts/InitializeDevelopmentEnvironment.psm1

```diff
@@ -427,19 +422,117 @@ function Invoke-PesterTests {
-        & (Get-RepositoryResolvedPath "logistics\scripts\run-smoke-tests.ps1")
+        & (Get-RepositoryResolvedPath "logistics/scripts/run-smoke-tests.ps1")
     }
 }
 
 function Invoke-DotnetTest {
     Invoke-Task -name $MyInvocation.MyCommand.Name -task {
-        & (Get-RepositoryResolvedPath "logistics\scripts\run-tests.ps1")
+        & (Get-RepositoryResolvedPath "logistics/scripts/run-tests.ps1")
+    }
+}
+
+function Get-DefaultNuGetProperties {
+    $buildConfiguration = 'debug'
+    if (-not [string]::IsNullOrWhiteSpace($env:msbuild_buildConfiguration)) { $buildConfiguration = $env:msbuild_buildConfiguration }
+
+    return @(
+        "authors=Ed-Fi Alliance"
+        "owners=Ed-Fi Alliance"
+        "configuration=$buildConfiguration"
+    )
+}
+
+function New-DatabasesPackage {
+    param(
+        [string] $ProjectPath,
+
+        [string] $PackageDefinitionFile = "$ProjectPath/$PackageId.nuspec",
+
+        [string] $PackageId,
+
+        [string] $Version,
+
+        [string[]] $Properties = @(),
+
+        [string] $OutputDirectory
+    )
+
+    Invoke-Task -name "$($MyInvocation.MyCommand.Name) ($(Split-Path $ProjectPath -Leaf))" -task {
+
+        Write-Host -ForegroundColor Magenta "& `"$ProjectPath/prep-package.ps1`" $PackageId"
+        & "$ProjectPath/prep-package.ps1" $PackageId
+        Write-Host
+
+        $nuget = Install-NuGetCli -ToolsPath $ToolsPath
+
+        $params = @{
+            PackageDefinitionFile = (Get-ChildItem "$ProjectPath/$PackageId.nuspec")
+            PackageId             = $PackageId
+            Version               = $Version
+            Properties            = $Properties
+            OutputDirectory       = $OutputDirectory
+            NuGet                 = $nuget
+        }
+        New-Package @params | Out-Host
+    }
+}
+
+function New-WebPackage {
+    param(
+        [string] $ProjectPath,
+
+        [string] $PackageDefinitionFile = "$ProjectPath/bin/**/**/publish/$(Split-Path $ProjectPath -Leaf).nuspec",
+
+        [string] $PackageId,
+
+        [string] $Version,
+
+        [string[]] $Properties = @(),
+
+        [string] $OutputDirectory`
+    )
+
+    Invoke-Task -name "$($MyInvocation.MyCommand.Name) ($(Split-Path $ProjectPath -Leaf))" -task {
+
+        $buildConfiguration = 'debug'
+        if (-not [string]::IsNullOrWhiteSpace($env:msbuild_buildConfiguration)) { $buildConfiguration = $env:msbuild_buildConfiguration }
+
+        $params = @(
+            "publish", $ProjectPath,
+            "--configuration", $buildConfiguration,
+            "--no-restore",
+            "--no-build"
+        )
+
+        Write-Host -ForegroundColor Magenta "& dotnet $params"
+        & dotnet $params | Out-Host
+        Write-Host
+
+        $PackageDefinitionFile = (Get-ChildItem $PackageDefinitionFile)
+        if (-not [string]::IsNullOrWhiteSpace($PackageId)) {
+            [xml] $xml = Get-Content $PackageDefinitionFile
+            $xml.package.metadata.id = $PackageId
```

Comment:
This seems suspect that we're modifying a source controlled file and changing it.  Why can't we pass the id along like you do above for the database package?  Do we need to update the checked in nuspec to have a variable (with a default here possibly) to allow better control of this?

Reply:
I think in other cases we're passing the `packageId` to `dotnet pack`, which works well, but here I'm using nuget.exe and I don't believe it allows you to parameterize the `id`. `dotnet pack` has the issue where you can't package something that isn't a project, specifically it requires at least a .csproj file, so it won't work for the databases package. I went with nuget because that's how it currently works with the build server. The build server also has a feature that is string replacing the id which I've recreated here.

I definitely would like to use `dotnet pack` for all of these in the future, but I went with reusing some of the functionality we already had.

Reply:
You can parameterize the id and pass it as a property, if you just put it as a variable in the nuspec, but it requires you to always pass the id in since it needs to be replaced.  Taking a quick look around, it looks like we've gotten away from that with most of the nuspecs, so we can defer on revisiting this if it matches for now.